DQMServices/Core/src/DQMStore.cc

```diff
@@ -2182,6 +2019,47 @@ DQMStore::postGlobalBeginLumi(const edm::GlobalContext &gc)
 //////////////////////////////////////////////////////////////////////
 //////////////////////////////////////////////////////////////////////
 //////////////////////////////////////////////////////////////////////
+/** Clone the lumisection-based histograms from the 'global' ones
+ * (which have lumi = 0) into per-lumi ones (with the lumi number)
+ * and reset the global ones.
+ * The per-lumi histograms can be saved by the output modules, and
+ * will be deleted at the beginninng of the next lumisection.
+ */
+
+void
+DQMStore::cloneLumiHistograms(uint32_t run, uint32_t lumi, uint32_t moduleId)
+{
+  if (verbose_ > 1) {
+    std::cout << "DQMStore::cloneLumiHistograms - Preparing lumi histograms for run: "
+              << run << ", lumi: " << lumi << ", module: " << moduleId << std::endl;
+  }
+
+  // acquire the global lock since this accesses the undelying data structure
+  std::lock_guard<std::mutex> guard(book_mutex_);
+
+  // MEs are sorted by (run, lumi, stream id, module id, directory, name)
+  // lumi deafults to 0
+  // stream id is always 0
+  std::string null_str("");
+  auto i = data_.lower_bound(MonitorElement(&null_str, null_str, run, moduleId));
+  auto e = data_.lower_bound(MonitorElement(&null_str, null_str, run, moduleId + 1));
+  for (; i != e; ++i) {
+    // handle only lumisection-based histograms
+    if (not LSbasedMode_ and not i->getLumiFlag())
+      continue;
+
+    // clone the lumisection-based histograms
+    MonitorElement clone{*i};
+    clone.globalize();
+    clone.setLumi(lumi);
+    clone.markToDelete();
+    data_.insert(std::move(clone));
+
+    // reset the ME for the next lumisection
+    const_cast<MonitorElement*>(&*i)->Reset();
```

Comment:
this is great, actually.

Can we remove DQMStore::postGlobalBeginLumi, https://github.com/cms-sw/cmssw/blob/master/DQMServices/Core/src/DQMStore.cc#L2159

Since it is now done here?

Reply:
Feel free to make a PR.

However, either that was already unnecessary (because of the call to `Reset()` in the `mergeAndResetMEsLuminositySummaryCache()` method) or is still needed (possibly for the legacy modules)